k8s/upgrades/0.5.x-0.6.0/README.md

```diff
@@ -0,0 +1,102 @@
+
+```
+#Delete older operator and storage classes. With OpenEBS 0.6, all the components are installed in namespace `openebs`
+# as opposed to `default` namespace in earlier releases. Before upgrading to 0.6, delete the older version and 
+# then apply the newer versions. 
+kubectl delete -f https://raw.githubusercontent.com/openebs/openebs/v0.5/k8s/openebs-operator.yaml
+kubectl delete -f https://raw.githubusercontent.com/openebs/openebs/v0.5/k8s/openebs-storageclasses.yaml
+#Wait for objects to be delete, you can check using `kubectl get deploy`
+
+#Install the 0.6 operator and storage classes.
+kubectl apply -f https://raw.githubusercontent.com/openebs/openebs/v0.6/k8s/openebs-operator.yaml
+kubectl apply -f https://raw.githubusercontent.com/openebs/openebs/v0.6/k8s/openebs-storageclasses.yaml
+```
+
+### Install/Upgrade using helm chart (using stable/openebs, openebs-charts repo, etc.,) 
+
+**The sample steps below will work if you have installed openebs with default values provided by stable/openebs helm chart.**
+
+- Run `helm ls` to get the release name of openebs. 
+- Upgrade using `helm upgrade -f https://openebs.github.io/charts/helm-values-0.6.0.yaml <release-name> stable/openebs`
+
+### Using customized operator YAML or helm chart.
+As a first step, you must update your custom helm chart or YAML with 0.6 release tags and changes made in the values/templates. 
+
+You can use the following as references to know about the changes in 0.6: 
+- stable/openebs [PR#6768](https://github.com/helm/charts/pull/6768) or 
+- openebs-charts [PR#1646](https://github.com/openebs/openebs/pull/1646) as reference.
+
+After updating the YAML or helm chart or helm chart values, you can use the above procedures to upgrade the OpenEBS Operator
+
+## Step 2: Upgrade the OpenEBS Volumes
+
+Even after the OpenEBS Operator has been upgraded to 0.6, the volumes will continue to work with 0.5.3 or 0.5.4. Each of the volumes should be upgraded (one at a time) to 0.6, using the steps provided below. 
+
+*Note: There has been a change in the way OpenEBS Controller Pods communicate with the Replica Pods. So, it is recommended to schedule a downtime for the application using the OpenEBS PV while performing this upgrade. Also, make sure you have taken a backup of the data before starting the below upgrade procedure.*
+
+In 0.5.x releases, when a replica is shutdown, it will get rescheduled to another availble node in the cluster and start copying the data from the other replicas. This is not a desired behaviour during upgrades, which will create new replica's as part of the rolling-upgrade. To pin the replicas or force them to the nodes where the data is already present, starting with 0.6 - we use the concept of nodeSelector and Tolerations that will make sure replica's are not moved on node or pod delete operations.
+
+So as part of upgrade, we recommend that you label the nodes where the replica pods are scheduled as follows:
```

Comment:
Discussed with @AmitKumarDas  about using node-selector with hostname vs a node-selector with generic label. With 0.6 we introduced the concept of a generic label, which is usually applied on 3 nodes. where the replica's need to be scheduled. one advantage of this is .. in case a node goes bad, a new node can be labelled with that generic storage label -- replica factor of 3 will be automatically configured. In case specific hostname labels are assigned, then all the replica deployments need to be patched. so going with generic label for now.